File: Children-Competition-C#/Service/ChildService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Concurs.Domain;
using Concurs.Repository;

namespace Concurs.Service
{
    internal class ChildService
    {
        private  ChildDBRepository childDBRepository;

        public ChildService(ChildDBRepository childDBRepository)
        {
            this.childDBRepository = childDBRepository;
        }

        public Child AddChild(Child child)
        {
            return this.childDBRepository.Save(child);
        }

        public Child FindChildByNameAge(string FirstName, string LastName, int age)
        {
            return this.childDBRepository.FindChildByNameAge(FirstName, LastName, age);
        }
    }
}

File: Children-Competition-Java/mpp-proiect-java-ABNegrea/Problema5/src/main/java/Concurs/Service/ChildService.java
package Concurs.Service;

import Concurs.Domain.Child;
import Concurs.Repository.ChildDBRepository;

public class ChildService {
    private ChildDBRepository childRepo;

    public ChildService(ChildDBRepository childRepo) {
        this.childRepo = childRepo;
    }

    public Child addChild(Child ch){
        return this.childRepo.save(ch);
    }

    public Child findChildByNameAge(String firstName, String lastName, int age)
    {
        return this.childRepo.findChildByNameAge(firstName,lastName,age);
    }
}
